examples/samplehold/ex_samplehold.cpp

```diff
@@ -0,0 +1,73 @@
+#include "daisysp.h"
+#include "daisy_seed.h"
+
+// Interleaved audio definitions
+#define LEFT (i)
+#define RIGHT (i + 1)
+
+using namespace daisysp;
+using namespace daisy;
+
+static DaisySeed  seed;
+static Oscillator osc;
+static Metro      tick;
+static WhiteNoise noise;
+static AdEnv      env;
+static SampleHold sh;
+
+float osc_out = 0;
+
+static void AudioCallback(float *in, float *out, size_t size)
+{
+    float osc_out, env_out;
+    for(size_t i = 0; i < size; i += 2)
+    {
+        // When the metro ticks, trigger the envelope to start.
+        if(tick.Process())
+        {
+            env.Trigger();
+            osc.SetFreq(sh.Sample(true, noise.Process() * 500 + 500));
+            tick.SetFreq(sh.Sample(false, 0) / 100 + 1);
+        }
```

Comment:
this is possibly a bit confusing to read.

might want to do something where you set a `trig` value and handle everything after

```C++
trig = tick.Process();
float newsamp = sh.Sample(trig, noise.Process() * 500 + 500);
if (trig)
{
    env.Trigger();
    tick.SetFreq(newsamp / 100 + 1);
}
osc.SetFreq(newsamp); // This could go in or outside of the if (trig) statement
```

The current code is confusing because you're calling Sample to get the sample without updating, but it reads as a bit confusing since the input source is different.